core/src/main/scala/sttp/tapir/Endpoint.scala

```diff
@@ -271,14 +275,14 @@ trait EndpointServerLogicOps[I, E, O, -R] { outer: Endpoint[I, E, O, R] =>
     * returned [[ServerEndpoint]] value (except for endpoint meta-data). To provide the logic in parts, see
     * [[serverLogicPart]] and [[serverLogicForCurrent]].
     */
-  def serverLogic[F[_]](f: I => F[Either[E, O]]): ServerEndpoint[I, E, O, R, F] = ServerEndpoint(this, _ => f)
+  def serverLogic[F[_]](f: I => F[Either[E, O]]): ServerEndpoint[R, F] = ServerEndpoint(this, _ => f)
```

Comment:
As discussed previously in #1156, can this method return `ServerEndpoint.Aux[I, E, O, R, F]`?